catalog/core/catalog-core-classification-impl/src/main/java/org/codice/alliance/catalog/core/internal/impl/classification/SecurityClassificationServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.catalog.core.internal.impl.classification;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.codice.alliance.catalog.core.internal.api.classification.SecurityClassificationService;
+
+/**
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public class SecurityClassificationServiceImpl implements SecurityClassificationService {
+
+    private Map<String, Integer> securityClassificationSortOrder = Collections.emptyMap();
+
+    @Override
+    public Comparator<String> getSecurityClassificationComparator() {
+        return new SecurityClassificationComparator(Collections.unmodifiableMap(
```

Comment:
If we create an instance of `SecurityClassificationComparator` with the map content when `setSortOrder()` is called and have the map inside the comparator only, then both the map and the comparator would be immutable, right? The only thing we may need to do is make the instance variable that holds the `SecurityClassificationComparator` `volatile` so it doesn't get re-assigned while it's being returned.